packages/@aws-cdk/pipelines/test/pipeline.test.ts

```diff
@@ -342,7 +342,7 @@ test('selfmutation stage correctly identifies nested assembly of pipeline stack'
 });
 
 test('selfmutation feature can be turned off', () => {
-  const stack = new Stack();
+  const stack = new Stack(app);
```

Comment:
If we don't pass `app` a default one gets created which does not use the modern synthesizer, and this results in the exception now